ipa-core/src/ff/boolean_array.rs

```diff
@@ -156,9 +156,100 @@ macro_rules! boolean_array_impl_small {
     };
 }
 
-//macro for implementing Boolean array, only works for a byte size for which Block is defined
+/// Macro to implement `Serializable` trait for boolean arrays. Depending on the size, conversion from [u8; N] to `BAXX`
+/// can be fallible (if N is not a multiple of 8) or infallible. This macro takes care of it and provides the correct
+/// implementation. Because macros can't do math, a hint is required to advise it which implementation it should provide.
+macro_rules! impl_serializable_trait {
+    ($name: ident, $bits: tt, fallible) => {
+        #[derive(thiserror::Error, Debug)]
+        #[error(
+            "The provided byte slice contains non-zero value(s) {0:?} in padding bits [{}..]",
+            $bits
+        )]
+        pub struct NonZeroPadding(Store);
```

Comment:
couldn't make it work for `BitSlice` because it is a DST and dealing with references in errors is something I'd want to avoid. But found a way to keep only one copy by using `GenericArray` instead.

wrt to macro hygiene, it does seem non-hygienic and I didn't even realize that. I changed it to propagate the `Store` type through the macro argument